src/main/java/org/camunda/community/benchmarks/StartDiScheduler.java

```diff
@@ -0,0 +1,229 @@
+        stats.hintOnNewPiPerSecondGoald(piPerSecondGoal);
+        if (diStartedGoal < 100) {
+            // we can handle this by starting one PI every x times 10 ms
+            batchSize = 1;
+            // better more than too less, then we can stop when we hit the limit
+            howOften = Math.round( Math.floor( 100.0 / piPerSecondGoal) );
+        } else {
+            // we need to start batches every 10 ms
+            howOften = 1;
+            // better more than too less, then we can stop when we hit the limit
+            batchSize = Math.round(Math.ceil( piPerSecondGoal / 100.0));
+        }
+        LOG.info("Configured benchmark to start " + piPerSecondGoal + " DIs per second. This means every " + howOften + ". interval (of the 100 x 10ms intervals in total) with a batch size of " + batchSize);
+    }
+
+    /**
+     * Every 10 ms we check how many process instances we need to start
+     * 10 ms is what the Java platform seems to be able to do reliably in different environments
+     * Add some initial delay to give the workers time to connect
+     */
+    //@Async
+    @Scheduled(fixedRate = 10, initialDelay = 5000)
+    public void startSomeProcessInstances() {
+      
+        long currentTime = System.currentTimeMillis();
+        long passedTime = currentTime - startTimeInMillis;
+
+        counter++;
+        long processInstancesToStart = 0;
+
+        // if we still have time till the second is up
+        if (passedTime < 1000) {
+            // Check if we should start another batch
+            if (counter % howOften == 0) {
+                // now check if we still want to do the full batch
+                if (diStarted + batchSize > diStartedGoal) {
+                    // just start the remaining instances
+                    processInstancesToStart = diStartedGoal - diStarted;
+                    diStarted += processInstancesToStart;
+                } else {
+                    // start the batch size
+                    processInstancesToStart = batchSize;
+                    diStarted += batchSize;
+                }
+            }
+        } else {
+            // check if we have remaining process instances to start
+            processInstancesToStart = diStartedGoal - diStarted;
+
+            // restart timer
+            LOG.debug("One second is over, resetting timer");
+            diStarted = 0;
+            startTimeInMillis = System.currentTimeMillis();
+        }
+        // start after all calculations to avoid that the next scheduler run intervenes.
+        // (the above calculations should always be faster than 10ms, starting a big batch might not)
+        // TODO: Think about if we should detect if starting takes longer than the 10ms interval
+        startDecisionInstances( processInstancesToStart );
+      
+    }
+
+    @Async
+    private void startDecisionInstances(long batchSize) {
+        for (int i = 0; i < batchSize; i++) {
+            executor.startDecisionInstance();
```

Comment:
If both executor implementations would have the same interface this code here could be generic